src/components/FiltersButton/FiltersButton.tsx

```diff
@@ -0,0 +1,58 @@
+import { useState } from 'react';
+import commonOptionButtonStyles from '../../common/commonOptionButton.module.css';
+import styles from './FiltersButton.module.css';
+
+const filterOptions: string[] = [
+   'Animals',
+   'Health',
+   'Education',
+   'Environment',
+];
+
+function FiltersButton() {
+   const [selectedFilters, setSelectedFilters] = useState<string[]>([]);
+   const [isOpen, setIsOpen] = useState<boolean>(false);
+
+   const toggleIsOpen = () => {
+      setIsOpen(!isOpen);
+   };
+
+   const handleFilterClick = (filter: string) => {
+      if (selectedFilters.includes(filter)) {
+         setSelectedFilters(selectedFilters.filter((f) => f !== filter));
+      } else {
+         setSelectedFilters([...selectedFilters, filter]);
+      }
+   };
+
+   const selected = (option: string) =>
+      selectedFilters.includes(option) ? styles.selected : '';
+
+   const opened = () => (isOpen ? commonOptionButtonStyles.open : '');
+
+   return (
+      <div className={styles.filters}>
+         <button
+            className={`${commonOptionButtonStyles.button} ${opened()}`}
+            onClick={toggleIsOpen}>
+            Filters: {selectedFilters.join(', ')} {isOpen ? '▲' : '▼'}
+         </button>
+         {isOpen && (
+            <ul className={commonOptionButtonStyles.optionsList}>
+               {filterOptions.map((option) => (
+                  <li
+                     key={option}
+                     className={`${
+                        commonOptionButtonStyles.optionItem
+                     } ${selected(option)}`}
+                     onClick={() => handleFilterClick(option)}>
+                     {selectedFilters.includes(option) ? `✓ ${option}` : option}
+                  </li>
+               ))}
+            </ul>
```

Comment:
As a general comment, the jsx should be as much rendering and as little logic and such.
If you can create a function with the logic and only put in the jsx the function, it would be more readable and easier to understand the rendered elements. 
You have here a few of those things, for example ```{selectedFilters.includes(option) ? `✓ ${option}` : option}```.